bootstrap/httpserver/fasthttp_server.go

```diff
@@ -0,0 +1,112 @@
+package httpserver
+
+import (
+	"time"
+
+	"github.com/orbs-network/membuffers/go"
+	"github.com/orbs-network/orbs-network-go/instrumentation/log"
+	"github.com/orbs-network/orbs-spec/types/go/protocol/client"
+	"github.com/orbs-network/orbs-spec/types/go/services"
+	"github.com/valyala/fasthttp"
+)
+
+type FastHttpServer interface {
+	GracefulShutdown(timeout time.Duration)
+}
+
+type fastHttpServer struct {
+	httpServer *fasthttp.Server
+	reporting  log.BasicLogger
+	publicApi  services.PublicApi
+}
+
+func NewFastHttpServer(address string, reporting log.BasicLogger, publicApi services.PublicApi) FastHttpServer {
+	server := &fastHttpServer{
+		reporting: reporting.For(log.String("adapter", "http-server")),
+		publicApi: publicApi,
+	}
+
+	readTimeout, _ := time.ParseDuration("500ms") //TODO error on failed parse
+	server.httpServer = &fasthttp.Server{
+		Handler:     server.createRouter(),
+		ReadTimeout: readTimeout,
+	}
+
+	go func() {
```

Comment:
I think we should pass context here so we could stop the server gracefully.

Reply:
What do you mean? the shutdown function waits for all connections to finish and then shuts down gracefully.